opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/instrumentor.py

```diff
@@ -26,38 +26,50 @@
 class BaseInstrumentor(ABC):
     """An ABC for instrumentors"""
 
-    def __init__(self):
-        self._is_instrumented = False
+    _instance = None
+    _is_instrumented = False
+
+    def __new__(cls):
+
+        if cls._instance is None:
+            cls._instance = object.__new__(cls)
+
+        return cls._instance
 
     @abstractmethod
-    def _instrument(self) -> None:
+    def _instrument(self, **kwargs) -> None:
         """Instrument"""
 
     @abstractmethod
-    def _uninstrument(self) -> None:
+    def _uninstrument(self, **kwargs) -> None:
         """Uninstrument"""
 
-    def instrument(self) -> None:
+    def instrument(self, **kwargs) -> None:
         """Instrument"""
 
         if not self._is_instrumented:
-            result = self._instrument()
+            result = self._instrument(**kwargs)
             self._is_instrumented = True
             return result
```

Comment:
I know if wasn't done in this PR by why is it returning if the annotation says it should return `None`?

Reply:
Removing annotation to adapt to all cases.